skins/oasis/modules/BodyController.class.php

```diff
@@ -80,247 +81,215 @@ public static function isGridLayoutEnabled() {
-					$railModuleList[1101] = array('FollowedPages', 'Index', null);
+			if ( $page_owner ) {
+				if ( !$page_owner->getGlobalPreference( 'hidefollowedpages' ) ) {
+					$railModuleList[1101] = [ 'FollowedPages', 'Index', null ];
 				}
 
-				if ( $wgEnableAchievementsExt ) {
-					$railModuleList[1102] = array('Achievements', 'Index', null);
+				if ( $this->wg->EnableAchievementsExt ) {
+					$railModuleList[1102] = [ 'Achievements', 'Index', null ];
 				}
 			}
 		}
 
-		if (self::isBlogPost() || self::isBlogListing()) {
-			$railModuleList[1250] = array('PopularBlogPosts', 'Index', null);
+		if ( self::isBlogPost() || self::isBlogListing() ) {
+			$railModuleList[1250] = [ 'PopularBlogPosts', 'Index', null ];
 		}
 
 		//  No rail on main page or edit page for oasis skin
 		// except &action=history of wall
-		if( !empty($wgEnableWallEngine) ) {
-			$isEditPage = !WallHelper::isWallNamespace($namespace) && BodyController::isEditPage() || $wgRequest->getVal('diff');
+		if ( !empty( $this->wg->EnableWallEngine ) ) {
+			$isEditPage = !WallHelper::isWallNamespace( $namespace ) && BodyController::isEditPage() || $this->wg->Request->getVal( 'diff' );
 		} else {
 			$isEditPage = BodyController::isEditPage();
 		}
 
 		if ( $isEditPage || WikiaPageType::isMainPage() ) {
-			$modules = array();
-			wfRunHooks( 'GetEditPageRailModuleList', array( &$modules ) );
-			wfProfileOut(__METHOD__);
+			$modules = [ ];
+			wfRunHooks( 'GetEditPageRailModuleList', [ &$modules ] );
 			return $modules;
 		}
 		// No modules on Custom namespaces, unless they are in the ContentNamespaces list, those get the content rail
-		if (is_array($wgExtraNamespacesLocal) && array_key_exists($subjectNamespace, $wgExtraNamespacesLocal) && !in_array($subjectNamespace, $wgContentNamespaces)) {
-			wfProfileOut(__METHOD__);
-			return array();
+		if ( is_array( $this->wg->ExtraNamespacesLocal ) && array_key_exists( $subjectNamespace, $this->wg->ExtraNamespacesLocal ) && !in_array( $subjectNamespace, $this->wg->ContentNamespaces ) ) {
+			return [ ];
 		}
 		// If the entire page is non readable due to permissions, don't display the rail either RT#75600
-		if (!$wgTitle->userCan( 'read' )) {
-			wfProfileOut(__METHOD__);
-			return array();
+		if ( !$this->wg->Title->userCan( 'read' ) ) {
+			return [ ];
 		}
 
-		$railModuleList[1440] = array('Ad', 'Index', ['slotName' => 'TOP_RIGHT_BOXAD']);
-		$railModuleList[1435] = array('AdEmptyContainer', 'Index', ['slotName' => 'NATIVE_TABOOLA_RAIL']);
-		$railModuleList[1100] = array('Ad', 'Index', ['slotName' => 'LEFT_SKYSCRAPER_2']);
-
-		unset($railModuleList[1450]);
+		$railModuleList[1440] = [ 'Ad', 'Index', [ 'slotName' => 'TOP_RIGHT_BOXAD' ] ];
+		$railModuleList[1435] = [ 'AdEmptyContainer', 'Index', [ 'slotName' => 'NATIVE_TABOOLA_RAIL' ] ];
+		$railModuleList[1100] = [ 'Ad', 'Index', [ 'slotName' => 'LEFT_SKYSCRAPER_2' ] ];
 
-		wfRunHooks( 'GetRailModuleList', array( &$railModuleList ) );
+		unset( $railModuleList[1450] );
```

Comment:
Please, add a comment and explain what it does. I mean, we know that it removes some item from the right rail, but more details would be appreciated.